graph/pkg/service/v0/drives.go

```diff
@@ -128,6 +134,43 @@ func (g Graph) GetRootDriveChildren(w http.ResponseWriter, r *http.Request) {
 	render.JSON(w, r, &listResponse{Value: files})
 }
 
+// CreateDrive creates a storage drive (space).
+func (g Graph) CreateDrive(w http.ResponseWriter, r *http.Request) {
+	us, ok := ctxpkg.ContextGetUser(r.Context())
+	if !ok {
+		errorcode.GeneralException.Render(w, r, http.StatusUnauthorized, "invalid user")
+		return
+	}
+
+	client, err := g.GetClient()
+	if err != nil {
+		errorcode.GeneralException.Render(w, r, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	spaceName := chi.URLParam(r, "drive-name")
+
+	csr := provider.CreateStorageSpaceRequest{
+		Owner: us,
+		Type:  "share",
+		Name:  spaceName,
+		Quota: &provider.Quota{
+			QuotaMaxBytes: 65536,
```

Comment:
This number is totally arbitrary. Since we don't really have defaults I would normally vote for yes, when we create a new role with the `create-space` permission it would be nice to also set the default quota and then use this one once we fetch it from the roles service. Such mechanism is not yet in place IIRC and it will blow the scope of the ticket out of proportions IMHO. I think this is a good first approach until we settle down the concept of roles and permissions.